source/fundamentals/crud/read-operations/count.txt

```diff
@@ -0,0 +1,210 @@
+===============
+Count Documents
+===============
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 1
+   :class: singlecol
+
+Overview
+--------
+
+In this guide, you can learn how to get an :ref:`accurate
+<accurate-count-go>` and :ref:`estimated <estimated-count-go>` count of
+the number of documents in your collection.
+
+Sample Data
+~~~~~~~~~~~
+
+To run the example in this guide, load the sample data into the
+``ratings`` collection of the ``tea`` database with the following
+snippet:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/CRUD/countAndEstimate.go
+   :language: go
+   :dedent:
+   :start-after: begin insert docs
+   :end-before: end insert docs
+
+.. include:: /includes/fundamentals/automatic-db-coll-creation.rst
+
+Each document contains a rating for a type of tea that corresponds to
+the ``type`` and ``rating`` fields.
+
+.. _accurate-count-go:
+
+Accurate Count
+--------------
+
+To count the number of documents that match your query filter, use the
+``CountDocuments()`` function.
+
+.. tip::
+
+   If you pass an empty query filter, this function returns the total
+   number of documents in the collection.
+
+Modify Behavior
+~~~~~~~~~~~~~~~
+
+You can modify the behavior of ``CountDocuments()`` by passing in a
+``CountOptions`` type. If you don't specify any options, the driver uses
+its default values.
+
+The ``CountOptions`` type allows you to configure options with the
+following functions:
+
+.. list-table::
+   :widths: 30 70
+   :header-rows: 1
+
+   * - Function
+     - Description
+
+   * - ``SetCollation()``
+     - | The type of language collation to use when sorting results.  
+       | Default: ``nil``
+
+   * - ``SetHint()`` 
+     - | The index to use to scan for documents to count. 
+       | Default: ``nil``
+
+   * - ``SetLimit()`` 
+     - | The maximum number of documents to count. 
+       | Default: ``0`` 
+
+   * - ``SetMaxTime()``
+     - | The maximum amount of time in milliseconds that the query can run on the server.
```

Comment:
```suggestion
     - | The maximum amount of time that the query can run on the server.
```

The argument is a `time.Duration`, so the caller can set any unit. [This example](https://github.com/mongodb/mongo-go-driver/blob/8a397a401000e6ba266e8a18beabc8af4da4a773/mongo/crud_examples_test.go#L270:L270) sets `2 * time.Second`.